examples/reflink_block.rs

```diff
@@ -21,9 +21,16 @@ fn main() -> std::io::Result<()> {
     to_file.set_len(len)?;
 
     let mut offset = 0u64;
-    while offset < len as u64 {
-        println!("reflink {offset}, {cluster_size}");
-        reflink_copy::ReflinkBlockBuilder::new(&from_file, &to_file, cluster_size)
+    while offset < len {
+        // Windows API clones the entire cluster regardless of the number of bytes actually used by
+        // the file in that cluster.
+        #[cfg(windows)]
+        let src_length = cluster_size;
+        #[cfg(not(windows))]
+        let src_length = NonZeroU64::new(cluster_size.get().min(len - offset)).unwrap();
```

Comment:
Two suggestions:
 - I think using `if cfg!(...)` is clearer, and given that it's a constant, rustc can trivially optimized them out if optimization is turned on
 - [`NonZero` actually supports `.min`](https://doc.rust-lang.org/std/num/type.NonZeroU64.html#impl-Ord-for-NonZero%3CT%3E) so the code can be simpler

```suggestion
        let src_length = if cfg!(windows) {
            // Windows API clones the entire cluster regardless of the number of bytes actually used by
            // the file in that cluster.
            cluster_size
        } else {
            cluster_size.min(NonZeroUsize::new(len - offset).unwrap())
        };
```

Reply:
thanks!